mvlearn/decomposition/groupica.py

```diff
@@ -0,0 +1,276 @@
+        )
+        X_pca = gpca.fit_transform(Xs)
+        self.grouppca_ = gpca
+        if self.solver == "fastica":
+            K, W, sources = fastica(
+                X_pca, **self.ica_kwargs, random_state=self.random_state
+            )
+        else:
+            K, W, sources = picard(
+                X_pca.T, **self.ica_kwargs, random_state=self.random_state
+            )
+            sources = sources.T
+        if K is not None:
+            self.components_ = np.dot(W, K)
+        else:
+            self.components_ = W
+        self.mixing_ = linalg.pinv(self.components_)
+        # Compute individual unmixing matrices by least-squares
+        self.individual_mixing_ = []
+        self.individual_components_ = []
+        sources_pinv = linalg.pinv(sources)
+        for X, mean in zip(Xs, self.means_):
+            lstq_solution = np.dot(sources_pinv, X - mean)
+            self.individual_components_.append(linalg.pinv(lstq_solution).T)
+            self.individual_mixing_.append(lstq_solution.T)
+        self.n_components_ = gpca.n_components_
+        self.n_features_ = gpca.n_features_
+        self.n_samples_ = gpca.n_samples_
+        self.n_views_ = gpca.n_views_
+        return self
+
+    def transform(self, Xs, y=None):
+        r"""
+        Transform the data Xs into sources.
+
+        Parameters
+        ----------
+        Xs: list of array-likes
+            - Xs shape: (n_views,)
+            - Xs[i] shape: (n_samples, n_features_i)
+        y : array, shape (n_samples,), optional
+
+        Returns
+        -------
+        X_transformed : list of array-likes or numpy.ndarray
+            The transformed data.
+            If `multiple_outputs` is True, it is a list with the estimated
+            individual sources.
+            If `multiple_outputs` is False, it is a single array containing the
+            shared sources.
+        """
+        check_is_fitted(self)
+        Xs = check_Xs(Xs, copy=True)
+
+        if self.multiple_outputs:
+            return [
+                np.dot(X - mean, W.T)
+                for W, X, mean in (
+                    zip(self.individual_components_, Xs, self.means_)
+                )
+            ]
+        else:
+            X = self.grouppca_.transform(Xs)
+            return np.dot(X, self.components_.T)
+
+    def inverse_transform(self, X_transformed):
+        r"""
+        Recover multiview data from transformed data.
+
+        Parameters
+        ----------
+        X_transformed : list of array-likes or numpy.ndarray
+            If `multiple_outputs` is True, it must be a list of arrays of shape
+            (n_samples, n_components) containing estimated sources.
+            If `multiple_outputs` is False, it must be a single
+            array containing shared sources.
+
+        Returns
+        -------
+        Xs : list of arrays
+            The recovered individual datasets.
+        """
+        check_is_fitted(self)
+
```

Comment:
I would check the input using `check_Xs` from [`mvlearn.utils`](https://github.com/mvlearn/mvlearn/blob/master/mvlearn/utils/utils.py#L19) to verify the input and it also converts a single view array to a list of that array so users can just pass in X which is a lot more user friendly.